modules/file-input/main/index.coffee

```diff
@@ -0,0 +1,165 @@
+            .on 'click', 'hx.file-input', -> dropdown.show()
+      else
+        selectedFiles.append 'span'
+          .text resolvedOptions.noFilesText or ''
+
+      @_.fileMap = fileMap
+
+      @emit 'change',
+        cause: 'user'
+        data: @_.fileMap.values()
+
+    fileListToMap = (fileList) ->
+      # Deals with duplicates if the file has the same UID
+      map = new hx.Map()
+      for file in fileList
+        fileUID = getFileUID file
+        map.set(fileUID, file)
+      map
+
+    if resolvedOptions.multiple and not options.buttonText
+      resolvedOptions.buttonText = 'Choose Files'
+
+    selection = hx.select @selector
+      .classed 'hx-file-input', true
+
+    input = hx.detached 'input'
+      .class 'hx-file-input-hidden'
+      .attr 'type', 'file'
+      .attr 'accept', resolvedOptions.accept
+      .attr 'multiple', if resolvedOptions.multiple then 'multiple' else undefined
+
+    group = hx.inputGroup()
+
+    button = hx.detached 'button'
+      .class "hx-file-input-button hx-btn #{resolvedOptions.buttonClass}"
+      .on 'click', -> input.node().click()
+      .add hx.detached('i').class('hx-file-input-icon hx-icon hx-icon-upload')
+      .add hx.detached('span').text resolvedOptions.buttonText
+
+    selectedFiles = hx.detached 'div'
+      .class 'hx-file-input-selected hx-section'
+      .text resolvedOptions.noFilesText or ''
+
+    dropdownDiv = hx.detached 'div'
+    setupDropdown = (element) ->
+      sel = hx.select element
+      self._.fileMap.values().map (file) ->
+        sel.append filePreview(file)
+    dropdown = new hx.Dropdown(dropdownDiv.node(), setupDropdown, {
+      ddClass: 'hx-file-input-dropdown'
+    })
+
+    input.on 'change', (e) -> handleFiles fileListToMap e.target.files
+
+    @_ =
+      options: resolvedOptions
+      input: input
+      button: button
+      fileMap: new hx.Map()
+
+    if resolvedOptions.dragEnabled
+      preventDefault = (e) ->
+        e.preventDefault()
+        e.stopPropagation()
+
+      drop = (e) =>
+        preventDefault(e)
+        unless @_.disabled
+          files = e.dataTransfer.files
+          validates = true
+          for file in files
+            break if not validates
+          if validates
+            handleFiles fileListToMap files
+
+      selection
+        .on 'dragenter', preventDefault
+        .on 'dragover', preventDefault
+        .on 'drop', drop
+
+    if resolvedOptions.disabled then @disabled(resolvedOptions.disabled)
+
+    selection
+      .add(input)
+      .add(group
+        .add(button)
+        .add(selectedFiles))
+      .add(dropdownDiv)
+
+  disabled: (disabled) ->
+    if arguments.length
+      disabledAttr = if disabled then 'disabled' else undefined
+      @_.input.attr 'disabled', disabledAttr
+      @_.button.attr 'disabled', disabledAttr
+      @_.disabled = disabled
+      this
+    else
+      @_.disabled or false
+
+  value: -> @_.fileMap.values()
```

Comment:
It returns the value of the file input as if they hadn't passed in a value. 
We could add a console warning but I didn't see the point warning the user that they've tried to do something that isn't documented.